common/src/main/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapter.java

```diff
@@ -0,0 +1,531 @@
+     * @param isFrt            True, if the supplied token is an FRT. False otherwise.
+     * @param refreshTokenPair The 'legacy' format pair -- the key & value.
+     * @return The resulting MicrosoftRefreshToken.
+     */
+    private MicrosoftRefreshToken createMicrosoftRefreshToken(@NonNull final MicrosoftAccount account,
+                                                              final boolean isFrt,
+                                                              @NonNull final Pair<String, ADALTokenCacheItem> refreshTokenPair) {
+        final String methodName = ":createMicrosoftRefreshToken";
+
+        try {
+            final String legacyCacheKey = refreshTokenPair.first;
+            final String rawRt = refreshTokenPair.second.getRefreshToken();
+            final String clientInfoStr = account.getClientInfo();
+            final ClientInfo clientInfo = new ClientInfo(clientInfoStr);
+            final String scope = "openid profile offline_access"; // default scopes
+            final String clientId = refreshTokenPair.second.getClientId();
+            final String environment = new URL(refreshTokenPair.second.getAuthority()).getHost();
+            String familyId = null;
+
+            if (isFrt) {
+                familyId = legacyCacheKey.substring(
+                        legacyCacheKey.lastIndexOf(FOCI_FLAG),
+                        legacyCacheKey.length()
+                );
+
+                familyId = familyId.replace(FOCI_FLAG, "");
+            }
+
+            return new MicrosoftRefreshToken(
+                    rawRt,
+                    clientInfo,
+                    scope,
+                    clientId,
+                    isFrt,
+                    environment,
+                    familyId
+            );
+        } catch (ServiceException | MalformedURLException e) {
+            final String errMsg = "Failed to create RefreshToken";
+            Logger.error(
+                    TAG + methodName,
+                    errMsg,
+                    null
+            );
+            Logger.errorPII(
+                    TAG + methodName,
+                    errMsg,
+                    e
+            );
+            return null;
+        }
+    }
+
+    /**
+     * Creates a {@link MicrosoftAccount} from the supplied {@link ADALTokenCacheItem}.
+     *
+     * @param refreshToken The credential used to derive the new account.
+     * @return The newly created MicrosoftAccount.
+     */
+    private MicrosoftAccount createAccount(@NonNull final ADALTokenCacheItem refreshToken) {
+        final String methodName = ":createAccount";
+        try {
+            final String rawIdToken = refreshToken.getRawIdToken();
+            final String uid = refreshToken.getUserInfo().getUserId();
+            final String utid = refreshToken.getTenantId();
+            final String environment = new URL(refreshToken.getAuthority()).getHost();
+
+            JsonObject clientInfo = new JsonObject();
```

Comment:
nit: can be final

Reply:
Thanks, resolved in d609ffdd0cb995eb7812e1a5370cd2bc8482f12e